### ISSUE TITLE
Release server 10.12.1

### ISSUE THREAD

**Issue by jnweiger** [owncloud/core#40721]
# ownCloud Server Minor Release Checklist

Most checklist items are meant to be actioned in the given order. In some cases tasks can be parallelized.
For some items, a fixed timing is important. E.g. customer communication is 2 weeks ahead of the release.
When raising tickets for specific sub-tasks, please add a link to said ticket by editing this release ticket.
Items that do not mention anybody are done by the BTR team members @jnweiger @gleikam

The progress of all release tickets is tracked in the releases board https://github.com/orgs/owncloud/projects/20 .
Whenever QA suggests a fix, the release ticket goes back to 'In dev'. The next beta/RC starts when it again reaches 'dev done/RC needed'.

Template: https://confluence.owncloud.com/display/OG/Releasing+ownCloud+Server

## Pre-Release Phase

- [ ] DEV: Paste this template into a github release ticket in owncloud/enterprise - @micbar
- [ ] DEV: Create a new GitHub Release project in ownCloud orga, move tickets there - @micbar
- [ ] DEV: Check status of last translation job https://drone.owncloud.com/owncloud/translation-sync/ - @dschmidt, @micbar, @jnweiger
- [ ] DEV: Check PHP dependency updates - @micbar @pdavis
- [ ] DEV: Check for any "[Test-only]" updates - @pdavis @individualIT
- [ ] DEV: [Create a preview of the changelog](https://confluence.owncloud.com/display/OG/Generating+Changelog+Entries) in core, post below in release ticket, highlight items related to clients - @micbar
- [ ] QA: Estimate QA efforts - @GeraldLeikam, @jnweiger, @jesmrec
- [ ] SECURITY
   - [ ] prepare advisories to be ready two weeks before estimated release - @IljaN
   - [ ] prepare cert update PR - @IljaN
   - [ ] check for vulnerable PHP dependencies - @xoxos
- [ ] BTR: Release kickoff meeting (Specify release schedule, roles, beta communication) - @micbar, @pako81, @jnweiger, @GeraldLeikam, @B3nWalter, @mmattel
- [ ] DOC: If minor release: schedule for later, to avoid backporting:
   - [ ] Prepare new owncloud/docs branch as explained in https://github.com/owncloud/docs-server/blob/master/docs/new-version-branch.md
   - [ ] PR for updating https://gitea.owncloud.services/devops/websites/src/branch/master/ansible/roles/docs/templates/config.j2
----
- [ ] COMMUNICATION: public announcement of start of beta phase, estimated timeline - @B3nWalter
- [ ] COMMUNICATION: Central posting: in a few days, expect beta. Please subscribe here
----
- [ ] BTR: Create testplan ticket from https://confluence.owncloud.com/display/OG/Server+OC10+Regression+Test+Plans
   - [ ] BTR: Add testplan sections for changelog testing - @jnweiger, @pako81
   - [ ] Highlight items: important, client related, mentioned in relnotes - @pako81

## Beta Phase (skip for patch releases)

- [ ] DEV: tag the beta in github (when public beta: create release-branch, bump version.php - else just tag beta on master) - @micbar
- [ ] BTR: [Build the beta](https://confluence.owncloud.com/display/OG/Releasing+ownCloud+Server#ReleasingownCloudServer-CommunityTarballs)
- [ ] BTR: Create [docker images](https://confluence.owncloud.com/display/OG/Creating+Docker+Image)
- [ ] DEV: Update Changelog in release ticket and testplan ticket (Highlight updates and new client related items) - @micbar, @jnweiger, @pako81
- [ ] COMMUNCATION: Organize Compatiblity Check with the web team  @kulmann
----
- [ ] COMMUNICATION: Ping Marketing: beta communication and links
- [ ] COMMUNICATION: Details - @B3nWalter
	- [ ] Add beta phase on /beta-testing webpage
	- [ ] Send newsletter to beta & testing subscribers list
	- [ ] Announce the release on central, [Category News, Tag server](https://central.owncloud.org/tags/c/news/server) - @jnweiger
	- [ ] Forward central link to eng + release-coordination mailing lists - @jnweiger
----
- [ ] QA: (_optional_) Changelog testing
- [ ] QA: (_optional_) Changelog testing against desktop client
- [ ] QA: (_optional_) Changelog testing against mobile clients - @jesmrec
----
- [ ] QA: Regression tests
   - [ ] If possible: Remove from testplan what is covered in CI (and update testplan template).
   - [ ] Checkmark items already covered during Changelog testing.
   - [ ] Do the remaining test items in the testplan ticket
----
- [ ] If "good enough":
	- [ ] BUILD: Build and publish Linux packages for beta in testing repo, then email to eng + release-coordination mailing lists
	- [ ] PUBLISH: Adjust [web update server] to deliver releases and test
		- [ ] Create or update PR to point web update server "beta" channel
		- [ ] PUBLISH: Deploy [web update server] to deliver releases - @crrodriguez
- [ ] MGMT: If code was changed: add (shortened) checklist section for next Beta here

## RC Phase

- [ ] SECURITY: Make sure [revocation list](https://github.com/owncloud/handbook/issues/76) for app keys have been merged - @micbar @iljaN
- [ ] SECURITY: Update [shipped ca-certificates](https://github.com/owncloud/handbook/issues/76) - @micbar @iljaN
- [ ] DEV: Create release branch `release-$version` from master (the last beta tag or later)
	- [ ] Feature freeze. All release-related fixes of to the release branch - @pako81
- [ ] DEV: [Move changelog/unreleased/* files and create preview](https://confluence.owncloud.com/display/OG/Generating+Changelog+Entries) in release branch - @pako81
	- [ ] Update Changelog in release ticket. (Highlight updates and new client related items) - @pako81
- [ ] DOCS: Provide technical information on which versions of the desktop sync client and mobile clients are compatible/recommended - @michaelstingl
- [ ] DOCS: Prepare, merge and deploy release notes - @pako81
- [ ] DOCS: Update version number in various places - @pako81
- [ ] BTR: [Build RC](https://confluence.owncloud.com/display/OG/Releasing+ownCloud+Server#ReleasingownCloudServer-CommunityTarballs)
- [ ] BTR: Create [docker images](https://confluence.owncloud.com/display/OG/Creating+Docker+Image)
----
- [ ] SECURITY: [check SonarCloud scanner](https://sonarcloud.io/dashboard?id=owncloud_core) - @micbar @IljaN
- [ ] SECURITY: [prepare advisories](https://github.com/owncloud/handbook/issues/77) - @micbar @IljaN
----
- [ ] COMMUNICATION: Ping Marketing: RC communication, feature freeze, and links.
- [ ] COMMUNICATION: Details - @B3nWalter
	- [ ] Add beta phase on /beta-testing webpage
	- [ ] Send newsletter to beta & testing subscribers list
	- [ ] Announce release (and feature freeze) on central, [Category News, Tag server](https://central.owncloud.org/tags/c/news/server)
	- [ ] Forward central link to eng + release-coordination mailing lists
----
- [ ] QA: Changelog testing (update testplan ticket with changelog items)
- [ ] QA: Changelog testing against desktop client
- [ ] QA: Changelog testing against mobile clients - @jesmrec
- [ ] QA: Smoke test owncloud-complete tarball
----
- [ ] MGMT: If "good enough":
	- [ ] BUILD: Build and publish [Linux packages](https://confluence.owncloud.com/display/OG/Create+OC10+Linux+Packages) for RC in testing repo
    - [ ] Email to eng + release-coordination mailing lists
	- [ ] PUBLISH: Adjust [web update server](https://confluence.owncloud.com/display/OG/Updating+the+OC10+web+updater) to deliver releases and test
		- [ ] Create or update PR to point web update server "beta" channel
		- [ ] QA: test update PR
		- [ ] PUBLISH: Deploy [web update server] to deliver releases - @crrodriguez
- [ ] MGMT: If code was changed: add (shortened) checklist section for next RC here

## Final Phase

- [ ] BTR: [Build final release](https://confluence.owncloud.com/display/OG/Releasing+ownCloud+Server#ReleasingownCloudServer-CommunityTarballs)
- [ ] QA: Smoke test owncloud-complete tarball
- [ ] QA: Smoke test desktop client
- [ ] QA: Smoke test mobile clients - @jesmrec
- [ ] PUBLISH: [Upload the tarballs for customers into portals](https://confluence.owncloud.com/display/OG/Releasing+ownCloud+Server#ReleasingownCloudServer-CommunityTarballs)
- [ ] PUBLISH: Update the link to the latest tarballs on the mirrors
   - [ ] run `gitea/appsigning/server/copy_to_latest.sh` # (do not copy manually, meta-files must be updated!)
- [ ] PUBLISH: Check automatic update "latest" link on https://doc.owncloud.com/server/latest/
- [ ] PUBLISH: check automated go.php links via https://github.com/owncloud/QA/blob/master/tools/core-test-doc-links.sh or similar
- [ ] PUBLISH: publish Linux packages
    - [ ] [Create Linux Packages](https://confluence.owncloud.com/display/OG/Create+OC10+Linux+Packages
	- [ ] Check repo keys: `bash` [check_opensuse_repo_keys.sh](https://github.com/owncloud/administration-internal/tools/check_opensuse_repo_keys.sh)
	- [ ] Check keyservers: `bash` [check_gpg_keyserver.sh](https://github.com/owncloud/administration-internal/tools/check_gpg_keyserver.sh) `4ABE1AC7557BEFF9`
	- [ ] Repo info to [portal.owncloud.com](https://portal.owncloud.com/apps/files/?dir=/Portal%20Data/All%20Account%20Data/ownCloud%20Enterprise%20Resources%20Data/Server&fileid=5661#editor)
	- [ ] Forward "packages ready" to eng + release-coordination mailing lists and channel
- [ ] BUILD, PUBLISH: Create [docker images](https://confluence.owncloud.com/display/OG/Creating+Docker+Image)
	- [ ] Setup demo instance for marketing screenshots.
- [ ] PUBLISH: Update maintenance and release schedule on customer.owncloud.com and https://github.com/owncloud/core/wiki/Maintenance-and-Release-Schedule - @pako81
- [ ] PR to merge back release branch. Check with @phil, if we want to keep the release-10.x.x branch -- if so branch a merge-back-10.x.x branch.
----
- [ ] COMMUNICATION: Release announcement on Central based on the blog entry [Category News, Tag server](https://central.owncloud.org/tags/c/news/server). New post, (link into last rc post)- @jnweiger
- [ ] COMMUNICATION: Announce release readiness on release-coordination mailing lists
- [ ] COMMUNICATION: Handover in marketing channel, links, versions, demo instance.  @jnweiger, @B3nWalter
	- [ ] Input for Customer Mailing: Main features, security fixes & advisories - @pako81, @IljaN, @Be3nWalter
- [ ] COMMUNICATION: All communications coordinated by marketing team. - @B3nWalter
	- [ ] Update download links on website for minimal and complete packages
		- [ ] Version number, release date
		- [ ] TAR md5, sha256, asc (full + minimal)
		- [ ] ZIP md5, sha256, asc (full + minimal)
		- [ ] ESX appliance
		- [ ] Virtual Box appliance
		- [ ] QCOW 2 appliance
		- [ ] VMWare appliance
----
- [ ] COMMUNICATION - @B3nWalter
	- [ ] Wikipedia + Wikidata update (de+en)
	- [ ] Blogpost about release
	- [ ] Social Media communications: Facebook, LinkedIn, Twitter
		- [ ] (_optional_) hackernews, reddit
----
- [ ] COMMUNICATION - @BB3nWalter
	- [ ] Enterprise: Customer Mailing
	- [ ] Website update feature table
	- [ ] Internal demo for ownCloud employees - @pako81
----
- [ ] PUBLISH: Adjust [web update server](https://confluence.owncloud.com/display/OG/Updating+the+OC10+web+updater) to deliver releases and test
	- [ ] Wait one week.
    - [ ] Study issues in central and core. Adjust web updater config
		- [ ] Point web update server "production/stable" channels for stable releases
		- [ ] Point the "beta" to the latest stable if there is no new beta/RC
			-> example ansible PR:
            - https://gitea.owncloud.services/devops/product/pulls/490
            - https://gitea.owncloud.services/devops/product/pulls/396
		- [ ] QA: test PR `curl -L  "http://updates.owncloud.com/server/?version=10x10x0x4x0x0xproductionxCommunityx"`
----
- [ ] APPLIANCE: release previous version of appliance
    - [ ] check status of previous version
	- [ ] [PR to owncloud-docker/appliance](https://github.com/owncloud-docker/appliance/pull/24/) - @jnweiger
    - [ ] Follow instructions in https://confluence.owncloud.com/display/OG/Appliance+Release
	- [ ] send appliance to univention for review - @voroyam @cdamken @jnweiger @Gleikam
    - [ ] test install in virtualbox
----
## Post-Release Phase

- [ ] Check https://doc.owncloud.com/docs/next/server_release_notes.html
- [ ] Schedule a retrospective call @michl19
- [ ] DEV: Close GitHub project @JanAckermann
- [ ] Ensure that the [release template](https://confluence.owncloud.com/display/OG/Releasing+ownCloud+Server) is up to date
- [ ] Ensure that the [testplan template](https://confluence.owncloud.com/display/OG/Server+OC10+Regression+Test+Plans) is up to date
- [ ] DEV: Close remaining items in GitHub Release project board @JanAckermann
- [ ] Check public presentation
   - [ ] downloads [en/de/es](https://owncloud.com/de/download-server),
   - [ ] check univention downloads,
   - [ ] check responses on central and social media
- [ ] Update https://github.com/owncloud/core/wiki/Maintenance-and-Release-Schedule
- [ ] Check that the changelog mentions the latest release: https://owncloud.com/changelog/server

**Comment by jnweiger** [comment]
### Changelog preview
Changelog for ownCloud Core [10.12.1] (2023-04-03)
=======================================
The following sections list the changes in ownCloud core 10.12.1 relevant to
ownCloud admins and users.

[10.12.1]: https://github.com/owncloud/core/compare/v10.12.0...v10.12.1

Summary
-------

* Bugfix - Respect User Home Folder Naming Rule home directory for chunks uploads: [#40693](https://github.com/owncloud/core/pull/40693)
* Bugfix - Add rewrite base to .htaccess: [#40696](https://github.com/owncloud/core/issues/40696)
* Bugfix - Prevent 507 Insufficient Storage on 32-bit systems: [#40567](https://github.com/owncloud/core/pull/40567)
* Change - Fix permission bits when enforcing passwords on public links: [#40701](https://github.com/owncloud/core/pull/40701)
* Change - Do not auto-enable user-key ecryption: [#40702](https://github.com/owncloud/core/pull/40702)
* Change - Update PHP dependencies: [#40691](https://github.com/owncloud/core/pull/40691)

Details
-------

* Bugfix - Respect User Home Folder Naming Rule home directory for chunks uploads: [#40693](https://github.com/owncloud/core/pull/40693)

   When using the User Home Folder Naming Rule (configurable in the Advanced tab of the LDAP
   wizard), which allows to specify the home folder by means of an LDAP attribute, chunks of users'
   uploads were wrongly created under the default data directory rather than inside the
   configured home directory. We are now using the getHome() method for getting the user's home so
   that chunks uploads respect the configured home directory.

   https://github.com/owncloud/core/pull/40693

* Bugfix - Add rewrite base to .htaccess: [#40696](https://github.com/owncloud/core/issues/40696)

   In previous core versions the rewrite base config.php option was not added to the generated
   .htaccess file. The use of a more hardened .htaccess file in version 10.12.0 (being introduced
   by https://github.com/owncloud/core/pull/40584) caused the files view in the web UI to be
   empty in URL via subfolder plus index.php-less setups. Additionally, the desktop app was not
   be able to sync and an error 405 (Method not allowed) was returned. Rewrite base is now correctly
   added to the .htaccess file.

   https://github.com/owncloud/core/issues/40696
   https://github.com/owncloud/core/pull/40697

* Bugfix - Prevent 507 Insufficient Storage on 32-bit systems: [#40567](https://github.com/owncloud/core/pull/40567)

   https://github.com/owncloud/core/pull/40567
   With
   the
   introduction
   of
   compatibility
   to
   32-bit
   systems
   broke
   as
   we
   are
   now
   casting
   $freeSpace
   to
   int
   and
   this
   caused
   an
   integer
   overflow
   on
   such
   systems
   when
   the
   free
   space
   was
   above
   the
   max
   supported
   value.
   We
   added
   therefore
   an
   additional
   check
   for
   32-bit
   systems
   in
   QuotaPlugin.php.

* Change - Fix permission bits when enforcing passwords on public links: [#40701](https://github.com/owncloud/core/pull/40701)

   It was not possible to enforce passwords on public link files with read + write permission. The
   admin can now check 'Enforce password protection for read + write + delete links' to do so.

   https://github.com/owncloud/core/issues/40699
   https://github.com/owncloud/core/pull/40701

* Change - Do not auto-enable user-key ecryption: [#40702](https://github.com/owncloud/core/pull/40702)

   Executing occ encryption:encrypt-all will no longer auto-enable user-key encryption.

   https://github.com/owncloud/enterprise/issues/4939
   https://github.com/owncloud/core/pull/40702
   https://doc.owncloud.com/docs/next/server_release_notes.html#deprecation-note-for-user-key-storage-encryption

* Change - Update PHP dependencies: [#40691](https://github.com/owncloud/core/pull/40691)

   The following have been updated: - guzzlehttp/psr7 (2.4.3 to 2.4.4) - icewind/streams (0.7.6
   to 0.7.7)

   The following have been updated in apps/files_external/3rdparty: - google/apiclient
   (2.12.6 to 2.13.1) - icewind/streams (0.7.6 to 0.7.7)

   https://github.com/owncloud/core/pull/40691
   https://github.com/owncloud/core/pull/40683
   https://github.com/owncloud/core/pull/40690

Changelog for ownCloud Core [10.12.0] (2023-02-24)
=======================================
The following sections list the changes in ownCloud core 10.12.0 relevant to
ownCloud admins and users.

[10.12.0]: https://github.com/owncloud/core/compare/v10.11.0...v10.12.0

Summary
-------

* Bugfix - Adjust installation database details: [#40348](https://github.com/owncloud/core/pull/40348)
* Bugfix - Bump bower_components/clipboard from v2.0.6 to v2.0.11 in /build: [#40064](https://github.com/owncloud/core/pull/40064)
* Bugfix - Properly remove file versions from the trashbin: [#40286](https://github.com/owncloud/core/issues/40286)
* Bugfix - "available for" in the mount point configuration will show displaynames: [#40412](https://github.com/owncloud/core/pull/40412)
* Bugfix - Skip public links when updating permissions of share's children: [#40420](https://github.com/owncloud/core/pull/40420)
* Bugfix - Add shib auth support for redirect url: [#40470](https://github.com/owncloud/core/pull/40470)
* Bugfix - Remove empty directories from the files_versions: [#40499](https://github.com/owncloud/core/pull/40499)
* Bugfix - Store checksums only if the whole stream has been read: [#40513](https://github.com/owncloud/core/pull/40513)
* Bugfix - Performance fix when deleting thumbnails: [#40514](https://github.com/owncloud/core/pull/40514)
* Bugfix - Bump minimatch from 3.0.4 to 3.1.2 in /build: [#40522](https://github.com/owncloud/core/pull/40522)
* Bugfix - Bump json5 from 2.2.0 to 2.2.3 in /build: [#40556](https://github.com/owncloud/core/pull/40556)
* Bugfix - Bump karma from 6.3.19 to 6.4.1 in /build: [#40558](https://github.com/owncloud/core/pull/40558)
* Bugfix - Bump moments.js from 2.29.1 to 2.29.4 in /build: [#40560](https://github.com/owncloud/core/pull/40560)
* Bugfix - Set length of oc_calendars.components to 255: [#40563](https://github.com/owncloud/core/pull/40563)
* Bugfix - Prevent creation of empty files/folders when no available quota: [#40567](https://github.com/owncloud/core/pull/40567)
* Bugfix - Bump underscore from 1.13.2 to 1.13.6 in /build: [#40568](https://github.com/owncloud/core/pull/40568)
* Bugfix - Fix the dav:cleanup-chunks command to work with a configured folder: [#40571](https://github.com/owncloud/core/pull/40571)
* Bugfix - Bump bower_components/showdown from 2.0.0 to 2.1.0 in /build: [#40579](https://github.com/owncloud/core/pull/40579)
* Bugfix - Fix orientation of images with exif data: [#40600](https://github.com/owncloud/core/pull/40600)
* Bugfix - Fix header title and claim rendered as escaped HTML: [#40605](https://github.com/owncloud/core/issues/40605)
* Bugfix - Use correct themed l10n app folder when app lives outside of server root: [#40607](https://github.com/owncloud/core/pull/40607)
* Bugfix - Fix share into share move scenario: [#40612](https://github.com/owncloud/core/pull/40612)
* Bugfix - Enable 2FA via provisioning API: [#40617](https://github.com/owncloud/core/issues/40617)
* Bugfix - Add index on oc_filecache: [#40633](https://github.com/owncloud/core/issues/40633)
* Bugfix - Adjust request body options for Guzzle7: [#40652](https://github.com/owncloud/core/pull/40652)
* Change - Allow specifying available space for objectstorages: [#40389](https://github.com/owncloud/core/pull/40389)
* Change - Drop PHP 7.3 support across the platform: [#40394](https://github.com/owncloud/core/pull/40394)
* Change - Test indirect resource existence: [#40406](https://github.com/owncloud/core/pull/40406)
* Change - Detect mime types of hidden files: [#40427](https://github.com/owncloud/core/pull/40427)
* Change - Copy the encryption keys first and then rename the files: [#40433](https://github.com/owncloud/core/pull/40433)
* Change - Drop unneeded Google SDK services: [#40444](https://github.com/owncloud/core/pull/40444)
* Change - Delete action is removed from sharing sections: [#40497](https://github.com/owncloud/core/pull/40497)
* Change - Allow to temporarily ignore invalid federated shares: [#40503](https://github.com/owncloud/core/pull/40503)
* Change - Update PHP dependencies: [#40337](https://github.com/owncloud/core/pull/40337)
* Change - Update Symfony components: [#40521](https://github.com/owncloud/core/pull/40521)
* Enhancement - Show WebDAV Url in personal setting under app passwords: [#40509](https://github.com/owncloud/core/pull/40509)
* Enhancement - Show username on personal profile page: [#40510](https://github.com/owncloud/core/pull/40510)
* Enhancement - Add legal privacy policy and imprint links to personal settings: [#40511](https://github.com/owncloud/core/pull/40511)
* Enhancement - Persistent major file version workflow: [#40531](https://github.com/owncloud/core/pull/40531)
* Enhancement - Add support for login policies: [#40574](https://github.com/owncloud/core/pull/40574)
* Enhancement - Add support for OCM via ScienceMesh: [#40577](https://github.com/owncloud/core/issues/40577)
* Enhancement - Add account creation time: [#2298](https://github.com/owncloud/enterprise/issues/2298)
* Enhancement - Improve UX on occ user:sync: [#40640](https://github.com/owncloud/core/pull/40640)
* Enhancement - Drag & Drop folders into public file upload: [#40643](https://github.com/owncloud/core/pull/40643)
* Enhancement - Make sender display name in mail notifications configurable: [#40671](https://github.com/owncloud/core/pull/40671)
* Enhancement - Tweak rewrite conditions in .htaccess: [#40584](https://github.com/owncloud/core/pull/40584)

Details
-------

* Bugfix - Adjust installation database details: [#40348](https://github.com/owncloud/core/pull/40348)

   The suggested host name and port syntax for the database host on the installation has been
   corrected.

   https://github.com/owncloud/core/issues/39871
   https://github.com/owncloud/core/pull/40348

* Bugfix - Bump bower_components/clipboard from v2.0.6 to v2.0.11 in /build: [#40064](https://github.com/owncloud/core/pull/40064)

   https://github.com/owncloud/core/pull/40064

* Bugfix - Properly remove file versions from the trashbin: [#40286](https://github.com/owncloud/core/issues/40286)

   Previously, restoring or removing a file from inside a folder that was deleted (so the folder
   and the contents are in the trashbin) didn't remove the versions of the file. Those versions
   were left in both the DB and the FS, taking space and degrading the performance.

   This is now being handled properly, so no additional resource is consumed due to the versions
   being left stranded.

   https://github.com/owncloud/core/issues/40286

* Bugfix - "available for" in the mount point configuration will show displaynames: [#40412](https://github.com/owncloud/core/pull/40412)

   The "available for" select of the mount configuration of external storages were using the
   group id. This wasn't a problem because for local groups the group id matches the group
   displayname, and for ldap groups the group id was the "cn" attribute. Due to recent changes, the
   ldap group will now use the objectuid attribute (or a similar attribute) as group id by default.
   This was causing the "available for" select to show that objectuid, so identifying the right
   group was problematic.

   Now, the "available for" select will show the group displayname, which for ldap is the "cn"
   attribute by default.

   Note that this happens on new installations. There is an automatic migration in place, so for
   upgrades, the "cn" attribute will be set as groupname in order to keep the old behavior

   https://github.com/owncloud/core/pull/40412

* Bugfix - Skip public links when updating permissions of share's children: [#40420](https://github.com/owncloud/core/pull/40420)

   Currently, updates to permissions of a share are wrongly propagated to public links children.
   This has now been fixed and public links are being skipped.

   https://github.com/owncloud/core/pull/40420

* Bugfix - Add shib auth support for redirect url: [#40470](https://github.com/owncloud/core/pull/40470)

   Without this change, server-side Apache Auth (e.g. Shibboleth) resulted in redirect to
   default owncloud page, instead of last visited page. We now correctly handle redirect_url.

   https://github.com/owncloud/enterprise/issues/5450
   https://github.com/owncloud/core/pull/40470
   https://github.com/owncloud/core/pull/40161

* Bugfix - Remove empty directories from the files_versions: [#40499](https://github.com/owncloud/core/pull/40499)

   Empty directories were left when the contained versions were deleted or moved. Large
   installations might end up with too many of these empty directories.

   Now, when a version is deleted, the containing directory will also be deleted if there aren't
   any more versions inside.

   https://github.com/owncloud/core/pull/40499

* Bugfix - Store checksums only if the whole stream has been read: [#40513](https://github.com/owncloud/core/pull/40513)

   Previously, range downloads (or downloads requesting a specific byte range) would store a
   checksum, if needed, based only on the requested range. This causes problems because the
   checksum is expected to be for the whole file.

   Now, those range downloads won't store a checksum because only a part of the file has been read,
   so the checksum would be incomplete.

   Some additional cases have been taken into account, mostly based on actions that could happen
   on the data stream, but they shouldn't happen normally.

   https://github.com/owncloud/core/pull/40513

* Bugfix - Performance fix when deleting thumbnails: [#40514](https://github.com/owncloud/core/pull/40514)

   Detecting unused thumbnails is now using a better optimized SQL statements which consumes
   less database and web server resources.

   https://github.com/owncloud/core/pull/40514

* Bugfix - Bump minimatch from 3.0.4 to 3.1.2 in /build: [#40522](https://github.com/owncloud/core/pull/40522)

   https://github.com/owncloud/core/pull/40522

* Bugfix - Bump json5 from 2.2.0 to 2.2.3 in /build: [#40556](https://github.com/owncloud/core/pull/40556)

   https://github.com/owncloud/core/pull/40556

* Bugfix - Bump karma from 6.3.19 to 6.4.1 in /build: [#40558](https://github.com/owncloud/core/pull/40558)

   https://github.com/owncloud/core/pull/40558

* Bugfix - Bump moments.js from 2.29.1 to 2.29.4 in /build: [#40560](https://github.com/owncloud/core/pull/40560)

   https://github.com/owncloud/core/pull/40560

* Bugfix - Set length of oc_calendars.components to 255: [#40563](https://github.com/owncloud/core/pull/40563)

   https://github.com/owncloud/core/issues/40537
   https://github.com/owncloud/core/pull/40563

* Bugfix - Prevent creation of empty files/folders when no available quota: [#40567](https://github.com/owncloud/core/pull/40567)

   Until now it was possible for users having 0 quota or who already reached the limit of their
   assigned quota to still create empty files/folders, which generates confusion. The PR fixes
   this behaviour.

   https://github.com/owncloud/enterprise/issues/5478
   https://github.com/owncloud/core/pull/40567

* Bugfix - Bump underscore from 1.13.2 to 1.13.6 in /build: [#40568](https://github.com/owncloud/core/pull/40568)

   https://github.com/owncloud/core/pull/40568

* Bugfix - Fix the dav:cleanup-chunks command to work with a configured folder: [#40571](https://github.com/owncloud/core/pull/40571)

   The ownCloud's FS was initialized partially to prevent contacting the LDAP server if it was
   configured. This was causing problems because the upload folder where the chunks were
   expected was a mountpoint, and due to the partial FS initialization such mountpoint was
   missing, so we were checking a different folder (the default one).

   Now, the ownCloud's FS will be fully initialized instead, so the mountpoint will be present and
   we check the right location.

   https://github.com/owncloud/core/pull/40571

* Bugfix - Bump bower_components/showdown from 2.0.0 to 2.1.0 in /build: [#40579](https://github.com/owncloud/core/pull/40579)

   https://github.com/owncloud/core/pull/40579

* Bugfix - Fix orientation of images with exif data: [#40600](https://github.com/owncloud/core/pull/40600)

   Some images with a large exif data had problems with the orientation when they were shown. This
   was caused by the native function failing to retrieve the exif data. Images with small exif data
   didn't have this problem.

   By making the chunk size of the stream bigger, the native function is able to load the exif data
   properly and return the information, and with such information we can fix the orientation of
   the image.

   https://github.com/owncloud/core/pull/40600

* Bugfix - Fix header title and claim rendered as escaped HTML: [#40605](https://github.com/owncloud/core/issues/40605)

   The files_sharing application template was escaping the HTML from the title and claim
   provided by the theme. This caused raw HTML to be displayed in the page header.

   https://github.com/owncloud/core/issues/40605
   https://github.com/owncloud/core/pull/40606

* Bugfix - Use correct themed l10n app folder when app lives outside of server root: [#40607](https://github.com/owncloud/core/pull/40607)

   When an app_path is pointing outside of the ownCloud server root or uses an symlink under
   certain conditions the l10n folder points to an invalid location and results in a crash of the
   server. This happened due to the assumption that app paths always start with the server root
   path.

   https://github.com/owncloud/core/pull/40607

* Bugfix - Fix share into share move scenario: [#40612](https://github.com/owncloud/core/pull/40612)

   Public links were lost upon moving share into another share as the share owner was not correctly
   set. This has been now partially fixed.

   https://github.com/owncloud/enterprise/issues/5565
   https://github.com/owncloud/core/pull/40612

* Bugfix - Enable 2FA via provisioning API: [#40617](https://github.com/owncloud/core/issues/40617)

   Two factor authentication can now be enabled using the provisioning api.

   https://github.com/owncloud/core/issues/40617

* Bugfix - Add index on oc_filecache: [#40633](https://github.com/owncloud/core/issues/40633)

   This index can help to speed-up bulk file operations.

   https://github.com/owncloud/core/issues/40633

* Bugfix - Adjust request body options for Guzzle7: [#40652](https://github.com/owncloud/core/pull/40652)

   In Guzzle major version 7 the body option of a request must be a string or similar. Requests that
   send arrays of items have been adjusted to use the form_params option. Developers of apps that
   use lib/private/Http/Client/IClient.php should check any calls to ensure that arrays of
   items are not passed in the body option. See the diffs of PHP docs in the linked PR for examples.

   https://github.com/owncloud/core/issues/40649
   https://github.com/owncloud/core/pull/40652

* Change - Allow specifying available space for objectstorages: [#40389](https://github.com/owncloud/core/pull/40389)

   Objectstorages are reporting only unknown storage space. This causes problems in other apps
   that rely on this storage method, e.g. metrics app that monitors available space. Now, new
   configuration in the storage level is added, allowing for using the system configuration
   variable by the apps or further extension of storage class for objectstorage.

   https://github.com/owncloud/core/issues/40665
   https://github.com/owncloud/enterprise/issues/5384
   https://github.com/owncloud/enterprise/issues/5006
   https://github.com/owncloud/core/pull/40389
   https://github.com/owncloud/core/pull/40669

* Change - Drop PHP 7.3 support across the platform: [#40394](https://github.com/owncloud/core/pull/40394)

   Support for security fixes for PHP 7.3 ended in Dec 2021 ownCloud core no longer supports PHP
   7.3. Ensure that you are using PHP 7.4.

   https://github.com/owncloud/core/pull/40394
   https://www.php.net/supported-versions.php

* Change - Test indirect resource existence: [#40406](https://github.com/owncloud/core/pull/40406)

   We now expect a not found error instead of permission denied error for some trash interactions.

   https://github.com/owncloud/core/pull/40406

* Change - Detect mime types of hidden files: [#40427](https://github.com/owncloud/core/pull/40427)

   Mime type of hidden files are now properly detected.

   https://github.com/owncloud/core/pull/40427

* Change - Copy the encryption keys first and then rename the files: [#40433](https://github.com/owncloud/core/pull/40433)

   Having encryption enabled, when a file was renamed, first the actual file was renamed, and then
   the encryption keys were moved to the new location. If something went wrong, it was possible
   that the keys weren't moved. This caused the file to become inaccessible because we couldn't
   decrypt the file due to the missing keys (which weren't in the right place)

   Now, when a file is renamed, the encryption keys will be copied first, and then the file will be
   renamed. If the encryption keys fail to be copied, the rename will fail. After the encryption
   keys are copied, the file could failed to be renamed. In this case, the copied keys will be
   removed, but the file will still be accessible because we still keep the old keys. The original
   keys (not the copy) will be removed if the file is renamed successfully.

   https://github.com/owncloud/core/pull/40433

* Change - Drop unneeded Google SDK services: [#40444](https://github.com/owncloud/core/pull/40444)

   Unused Google SDK services are removed to reduce package size.

   https://github.com/owncloud/core/pull/40444

* Change - Delete action is removed from sharing sections: [#40497](https://github.com/owncloud/core/pull/40497)

   In the files apps, the "shared with others" and "shared by link" sections allowed people to use a
   delete action on a file or folder present in that list. This was causing problems because people
   accidentally removed the folder when, in fact, they wanted to unshare it.

   This delete action isn't present any longer. You can unshare from those views by accessing the
   file or folder's details. If you want to delete the file or folder, you can do it from the regular
   "all files" section.

   https://github.com/owncloud/core/pull/40497

* Change - Allow to temporarily ignore invalid federated shares: [#40503](https://github.com/owncloud/core/pull/40503)

   This change is targeted mostly at tightly federated setups

   Currently, if federated share is invalid or api endpoint returns not found, availability
   check would validate whether this is a problem with a server and if checks complete that given
   share is removed. However, in some cases these checks might not be enough (e.g. complex
   migrations in tightly federated setups), and in that case invalidation behaviour can be
   disabled using below app setting:

   Files_sharing.enable_cleanup_invalid_external_shares='no'

   https://github.com/owncloud/enterprise/issues/5427
   https://github.com/owncloud/core/pull/40503

* Change - Update PHP dependencies: [#40337](https://github.com/owncloud/core/pull/40337)

   The following have been updated: - doctrine/event-manager (1.1.2 to 1.2.0) -
   doctrine/instantiator (1.4.1 to 1.5.0) - doctrine/lexer (1.2.3 to 2.1.0) -
   egulias/email-validator (3.2.1 to 3.2.5) - guzzlehttp/guzzle (7.4.5 to 7.5.0) -
   guzzlehttp/promises (1.5.1 to 1.5.2) - guzzlehttp/psr7 (2.4.0 to 2.4.3) -
   phpseclib/phpseclib (3.0.14 to 3.0.18) - laminas/laminas-filter (2.12.0 to 2.22.0) -
   laminas/laminas-inputfilter (2.12.1 to 2.21.0) - laminas/laminas-servicemanager (3.7.0
   to 3.17.0) - laminas/laminas-stdlib (3.11.0 to 3.13.0) - laminas/laminas-validator
   (2.19.0 to 2.25.0) - league/flysystem (1.1.9 to 1.1.10) - phpseclib/phpseclib (3.0.18 tp
   3.0.19) - psr/container (1.1.1 to 1.1.2) - punic/punic (3.7.0 to 3.8.0) - sabre/uri (2.2.3 to
   2.3.2) - sabre/vobject (4.5.0 to 4.5.3)

   The following have been updated in apps/files_external/3rdparty: - google/auth (v1.21.1 to
   v1.23.0) - google/apiclient-services (v0.259.0 to v0.272.0) - guzzlehttp/psr7 (2.4.0 to
   2.4.3)

   https://github.com/owncloud/core/pull/40337
   https://github.com/owncloud/core/pull/40394
   https://github.com/owncloud/core/pull/40410
   https://github.com/owncloud/core/pull/40424
   https://github.com/owncloud/core/pull/40448
   https://github.com/owncloud/core/pull/40449
   https://github.com/owncloud/core/pull/40494
   https://github.com/owncloud/core/pull/40543
   https://github.com/owncloud/core/pull/40554
   https://github.com/owncloud/core/pull/40568
   https://github.com/owncloud/core/pull/40591
   https://github.com/owncloud/core/pull/40668

* Change - Update Symfony components: [#40521](https://github.com/owncloud/core/pull/40521)

   The following Symfony components have been updated to: - console 4.4.49 - translation 4.4.47

   The following Symfony polyfill components have been updated: - symfony/polyfill-iconv
   (v1.26.0 to v1.27.0) - symfony/polyfill-intl-idn (v1.26.0 to v1.27.0) -
   symfony/polyfill-intl-normalizer (v1.26.0 to v1.27.0) - symfony/polyfill-mbstring
   (v1.26.0 to v1.27.0) - symfony/polyfill-php72 (v1.26.0 to v1.27.0) -
   symfony/polyfill-php73 (v1.26.0 to v1.27.0) - symfony/polyfill-php80 (v1.26.0 to
   v1.27.0)

   Https://symfony.com/blog/symfony-4-4-45-released
   https://github.com/owncloud/core/pull/40337
   https://symfony.com/blog/symfony-4-4-47-released
   https://github.com/owncloud/core/pull/40424
   https://symfony.com/blog/symfony-4-4-48-released
   https://github.com/owncloud/core/pull/40448
   https://symfony.com/blog/symfony-4-4-49-released
   https://github.com/owncloud/core/pull/40517

   Code that has been deprecated in Symfony 4 has been refactored to be ready for Symfony 5.

   https://github.com/owncloud/core/pull/40521
   https://github.com/owncloud/core/pull/40575
   https://github.com/owncloud/core/pull/40592

* Enhancement - Show WebDAV Url in personal setting under app passwords: [#40509](https://github.com/owncloud/core/pull/40509)

   For easy access of files through WebDAV the url is displayed right under the app password
   section.

   https://github.com/owncloud/core/pull/40509

* Enhancement - Show username on personal profile page: [#40510](https://github.com/owncloud/core/pull/40510)

   The username as well as the full name of a user is now shown on their personal general settings
   page.

   https://github.com/owncloud/core/pull/40510

* Enhancement - Add legal privacy policy and imprint links to personal settings: [#40511](https://github.com/owncloud/core/pull/40511)

   The links for legal.privacy_policy_url and legal.imprint_url are now displayed on the
   personal general settings page so that they are conveniently available for all users to see.
   These are only displayed if they are set.

   https://github.com/owncloud/core/pull/40511

* Enhancement - Persistent major file version workflow: [#40531](https://github.com/owncloud/core/pull/40531)

   - Restore operation logic changed. Now restore is creating new current version of the file from
   one of past noncurrent versions of the file. Current version also receives incremented mtime
   for the file, and author of the files is the user that restored the file. The old noncurrent
   version is no longer removed upon restore and current version no longer receives mtime of the
   version. - The current version of the file is now shown in the Versions Tab, highlighted with
   "gray" background - Versions now persist additional extended metadata on versioning tags,
   that allow easier identification of the versions. Each update increases minor version for the
   file. Current version of the file now can be published, which increases major version tag. -
   Each new edit of the file would create noncurrent versions. The ones tagged with major version
   due to publishing, will be persisted long term and wont be subject to any retention policies. -
   Migrate from deprecated save_version_author to save_version_metadata

   https://github.com/owncloud/enterprise/issues/5286
   https://github.com/owncloud/core/pull/40531
   https://github.com/owncloud/core/pull/40641

* Enhancement - Add support for login policies: [#40574](https://github.com/owncloud/core/pull/40574)

   Support for login policies has been added in order to block the login of users under some
   circumstances. By default, there isn't any restriction, so any user can login normally
   (assuming the password is correct)

   A group login policy has been added. This policy allows or denies the user from login based on the
   login type being used by the user (username + password, openidconnect, etc) and whether he
   belongs to specific groups. This can be used to ensure a group of users are always authenticated
   using a determined authentication mechanism.

   https://github.com/owncloud/core/pull/40574

* Enhancement - Add support for OCM via ScienceMesh: [#40577](https://github.com/owncloud/core/issues/40577)

   We've added an if-statement in the files_sharing ShareesController code that searches for
   remote sharees. When the config entry `sharing.remoteShareesSearch` is set to the name of a
   class that is registered in the server container and that implements
   `IRemoteShareesSearch`, for instance the 'ScienceMeshSearchPlugin' that the
   'sciencemesh' app registers, use it instead of the federatedfilesharing app to find sharee
   matches for OCM sharing.

   https://github.com/owncloud/core/issues/40577
   https://github.com/pondersource/oc-sciencemesh/pull/39

* Enhancement - Add account creation time: [#2298](https://github.com/owncloud/enterprise/issues/2298)

   Adding account creation time in oc_accounts table

   https://github.com/owncloud/enterprise/issues/2298
   https://github.com/owncloud/core/pull/40588

* Enhancement - Improve UX on occ user:sync: [#40640](https://github.com/owncloud/core/pull/40640)

   Backend class aliases have been added to improve usability of this command.

   https://github.com/owncloud/core/pull/40640

* Enhancement - Drag & Drop folders into public file upload: [#40643](https://github.com/owncloud/core/pull/40643)

   Previously only files were accepted via drag & drop. Users can now also drag folders into a
   public link that has the filedrop flag. When adding a folder, the hierarchy is flattened out and
   all files are added without any subfolders. Name collisons are avoided as usual.

   https://github.com/owncloud/enterprise/issues/5489
   https://github.com/owncloud/core/pull/40643

* Enhancement - Make sender display name in mail notifications configurable: [#40671](https://github.com/owncloud/core/pull/40671)

   In some cases mail notifications related to sharing activities are blocked by mail filters as
   they are flagged as email impersonation. In such cases it may be desirable for an oC admin to have
   a config option for removing the sender display name from the "From" address. This is now
   possible by setting the following config.php parameter:

   Remove_sender_display_name => true

   https://github.com/owncloud/core/pull/40671

* Enhancement - Tweak rewrite conditions in .htaccess: [#40584](https://github.com/owncloud/core/pull/40584)

   Changed the RewriteCond rules in the `.htaccess` file to match the expected paths.

   https://github.com/owncloud/core/pull/40584

Changelog for ownCloud Core [10.11.0] (2022-08-23)

**Comment by pako81** [comment]
The release notes contain slightly different links for the mentioned issues. I guess we should stay consistent.

- Bugfix - Respect User Home Folder Naming Rule home directory for chunks uploads --> https://github.com/owncloud/core/pull/40693 and https://github.com/owncloud/core/pull/40719

- Bugfix - Add rewrite base to .htaccess --> https://github.com/owncloud/core/pull/40697

- Bugfix - Prevent 507 Insufficient Storage on 32-bit systems --> https://github.com/owncloud/core/pull/40709

- Change - Fix permission bits when enforcing passwords on public links --> correct

**Comment by jnweiger** [comment]
Yes, thanks! Done: https://github.com/owncloud/core/commit/39262545c987453fe42afc9c8d4221a73cb6335e

**Comment by ho4ho** [comment]
Look like template used by creating issue many problems:

1. `BTR team members @jnweiger @gleikam`: One user `@gleikam` not knowing by GitHub
2. `Check for any "[Test-only]" updates - @pdavis @individualIT`: One user `@individualIT` not knowing by GitHub
3. `@Be3nWalter`: `@B3nWalter`
4. `@BB3nWalter`: `@B3nWalter`
5. `for review - @voroyam @cdamken @jnweiger @Gleikam`: `@GeraldLeikam`
6. `Check with @phil, if we want`: Looking not like ownCloud people

**Comment by jnweiger** [comment]
> Look like template used by creating issue many problems:
> 
>     1. `BTR team members @jnweiger @gleikam`: One user `@gleikam` not knowing by GitHub
> 
>     2. `Check for any "[Test-only]" updates - @pdavis @individualIT`: One user `@individualIT` not knowing by GitHub
> 
>     3. `@Be3nWalter`: `@B3nWalter`
> 
>     4. `@BB3nWalter`: `@B3nWalter`
> 
>     5. `for review - @voroyam @cdamken @jnweiger @Gleikam`: `@GeraldLeikam`
> 
>     6. `Check with @phil, if we want`: Looking not like ownCloud people

Yes, the template is maintained in an internal tool, where the usernames are different. We theoretically, we can remove all the @ before the names -- I need to ping people manually anyways :-)

**Comment by mmattel** [comment]
closing?

**Comment by voroyam** [comment]
![image](https://user-images.githubusercontent.com/24212810/233339601-2e0a06f1-a7a0-4b76-b0ed-31c690905e67.png)

why have so many tasks in the template if many remain unchecked?
